rust/src/lib/ovsdb/global_conf.rs

```diff
@@ -33,46 +36,200 @@ fn convert_map(
+            Err(e)
         }
-        row.insert(
-            "external_ids".to_string(),
-            Value::Array(vec![
-                Value::String("map".to_string()),
-                Value::Array(value_array),
+    }
+}
+
+fn append_purge_action(operations: &mut Vec<OvsDbOperation>, column: &str) {
+    log::info!("Purging OVS table {GLOBAL_CONFIG_TABLE} table column {column}");
+    let mut row = HashMap::new();
+    row.insert(
+        column.to_string(),
+        Value::Array(vec![
+            Value::String("map".to_string()),
+            Value::Array(Vec::new()),
+        ]),
+    );
+    operations.push(OvsDbOperation::Update(OvsDbUpdate {
+        table: GLOBAL_CONFIG_TABLE.to_string(),
+        conditions: Vec::new(),
+        row,
+    }))
+}
+
+// Use `delete` mutator for Value set to None, otherwise use `insert`.
+fn append_mutations(
+    mutations: &mut Vec<OvsDbMutation>,
+    column: &str,
+    data: &HashMap<String, Option<String>>,
+) {
+    // All key should be deleted first because `insert` does not override
+    // existing values.
+    let delete_keys: Vec<Value> = data
+        .keys()
+        .map(|k| {
+            log::debug!("Removing old value of key {} from {}", k, column);
+            Value::String(k.to_string())
+        })
+        .collect();
+    if !delete_keys.is_empty() {
+        mutations.push(OvsDbMutation {
+            column: column.to_string(),
+            mutator: "delete".to_string(),
+            value: Value::Array(vec![
+                Value::String("set".to_string()),
+                Value::Array(delete_keys),
             ]),
-        );
-        let mut value_array = Vec::new();
-        for (k, v) in ovs_conf.other_config.iter() {
-            if let Some(v) = v {
-                value_array.push(Value::Array(vec![
+        })
+    }
+    let insert_values: Vec<Value> = data
+        .iter()
+        .filter_map(|(k, v)| {
+            v.as_ref().map(|v| {
+                log::info!("Inserting key {} value {} into {}", k, v, column);
+                Value::Array(vec![
                     Value::String(k.to_string()),
                     Value::String(v.to_string()),
-                ]));
+                ])
+            })
+        })
+        .collect();
+    if !insert_values.is_empty() {
+        mutations.push(OvsDbMutation {
+            column: column.to_string(),
+            mutator: "insert".to_string(),
+            value: Value::Array(vec![
+                Value::String("map".to_string()),
+                Value::Array(insert_values),
+            ]),
+        })
+    }
+}
+
+pub(crate) fn ovsdb_apply_global_conf(
+    merged_state: &MergedNetworkState,
+) -> Result<(), NmstateError> {
+    if !merged_state.ovsdb.is_changed() && !merged_state.ovn.is_changed() {
+        log::debug!("No OVSDB changes");
+        return Ok(());
+    }
+
+    let mut cli = OvsDbConnection::new()?;
+    let mut operations = Vec::new();
+    let mut is_external_ids_purged = false;
```

Comment:
you use this variable to control if you purge *both* the `external_ids` **and** the `other_config` tables.

Could we name this something a bit more generic ? e.g. `is_global_table_purged` or something like that ? 

This is a nit you can ignore.

Reply:
No. Only `external_ids` purged require OVN reinstall. The `is_external_ids_purged` is true when full purging or `external_ids` is set to purging.